astrocut/make_cube.py

```diff
@@ -346,3 +349,500 @@ def make_cube(self, file_list, cube_file="img-cube.fits", sector=None, max_memor
+        """
+
+        with fits.open(self.template_file, mode='denywrite', memmap=True) as ffi_data:
+            
+            # The image specific header information will be saved in a table in the second extension
+            secondary_header = ffi_data[0].header
+
+            # set up the image info table
+            already_there = []
+            cols = []
+            for kwd, val, cmt in secondary_header.cards: 
+                if type(val) == str:  
+                    tpe = "S" + str(len(val))  # TODO: Maybe switch to U?
+                elif type(val) == int:
+                    tpe = np.int32
+                else:
+                    tpe = np.float64
+                    
+                length = len(self.file_list)
+                if kwd not in already_there:
+                    already_there.append(kwd)
+                else: 
+                    continue
+                cols.append(Column(name=kwd, dtype=tpe, length=length, meta={"comment": cmt}))
+
+            cols.append(Column(name="FFI_FILE", dtype="S" + str(len(os.path.basename(self.template_file))),
+                               length=length))
+            
+            self.info_table = Table(cols)
+
+    def _build_cube_file(self, cube_file):
+        """ Build the cube file on disk with primary header, cube extension header,
+        and space for the cube, filled with zeros.
+
+        Note, this will overwrite the file if it already exists.
+
+        Parameters
+        ----------
+        cube_file : str 
+            Optional. The filename/path to save the output cube in. 
+        """
+        
+        # Making sure the output directory exists
+        direc, _ = os.path.split(cube_file)
+        if direc and not os.path.exists(direc):
+            os.makedirs(direc)
+
+        # Writing the primary header
+        hdu0 = fits.PrimaryHDU(header=self.primary_header)
+        hdul = fits.HDUList([hdu0])
+        hdul.writeto(cube_file, overwrite=True)
+
+        # Making the cube header and writing it
+        data = np.zeros((100, 100, 10, 2), dtype=np.float32)
+        hdu = fits.ImageHDU(data)
+        header = hdu.header
+        header["NAXIS4"], header["NAXIS3"], header["NAXIS2"], header["NAXIS1"] = self.cube_shape
+
+        # I think this part increases the buffer size of ext1 and writes 
+        # the cube in there?
+        with open(cube_file, 'ab') as CUBE:
+            CUBE.write(bytearray(header.tostring(), encoding="utf-8"))
+
+        # Expanding the file to fit the full data cube
+        # fits requires all blocks to be a multiple of 2880
+        cubesize_in_bytes = ((np.prod(self.cube_shape) * 4 + 2880 - 1) // 2880) * 2880
+        filelen = os.path.getsize(cube_file)
+        
+        with open(cube_file, 'r+b') as CUBE:
+            # What is this .seek?
+            # Looks like it's expanding the buffer size of the file?
+            CUBE.seek(filelen + cubesize_in_bytes - 1)
+            # b for buffer?
```

Comment:
the `b` prefix signifies a "Byte literal". i think this comment and the ones above should be removed. can possibly replace with ...

`# seek to end of file and write null byte`